### PR TITLE
[NEW] Rewrite Apps

### PR DESCRIPTION
**Pull request by gabriellsh**
closes https://github.com/RocketChat/Rocket.Chat/issues/16471
closes https://github.com/RocketChat/Rocket.Chat/pull/16596
## Screenshots
image
image

## Types of changes

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] Improvement (non-breaking change which improves a current function)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Hotfix (a major bugfix that has to be merged asap)
- [ ] Documentation Update (if none of the other choices apply)

## Checklist

- [x] I have read the [CONTRIBUTING](https://github.com/RocketChat/Rocket.Chat/blob/develop/.github/CONTRIBUTING.md#contributing-to-rocketchat) doc
- [x] I have signed the [CLA](https://cla-assistant.io/RocketChat/Rocket.Chat)
- [x] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works (if applicable)
- [ ] I have added necessary documentation (if applicable)
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: client/admin/apps/AppsTable.js [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+import AppStatus from './AppStatus';
+import { GenericTable, Th } from '../../components/GenericTable';
+import { useLoggedInCloud } from './hooks/useLoggedInCloud';
+import { useTranslation } from '../../contexts/TranslationContext';
+import { useRoute } from '../../contexts/RouterContext';
+import { useFilteredLocalApps } from './hooks/useFilteredLocalApps';
+import { useResizeInlineBreakpoint } from '../../hooks/useResizeInlineBreakpoint';
+import AppMenu from './AppMenu';
+
+const FilterByText = React.memo(({ setFilter, ...props }) => {
```
Comment (ggazzo) [inline client/admin/apps/AppsTable.js:15]: I think we have a o lot of them... should we reuse? maybe create under `GenericTable`

Comment (gabriellsh) [reply client/admin/apps/AppsTable.js:15]: Yes definitely. This is part of a conversation we had a few days ago, that we would need to replace this for most tables. But I think it should be done on a separate PR so we can unify all of 'em.

File: client/admin/apps/hooks/useAppInfo.js [hunk 2]
```diff
@@ -17,20 +17,60 @@ const getBundledIn = async (appId, appVersion) => {
...
+	} catch (e) {
+		handleAPIError(e);
+	}
+};
+
+const getApis = async (appId, installed) => {
+	if (!installed) { return {}; }
+	try {
+		const apis = await Apps.getAppApis(appId);
+		return apis;
```
Comment (ggazzo) [inline client/admin/apps/hooks/useAppInfo.js:34]:
```suggestion
		return await Apps.getAppApis(appId);
```

**Comment by tassoevan** [comment]
Jun-17-2020 20-56-05

**Pull request merged**